### ISSUE TITLE
A new website for Gin 🩵

### ISSUE THREAD

**Issue by imlargo** [gin-gonic/website#288]
Hi everyone! 

For such an important and widely used framework as Gin, it would be great if the website matched its relevance and potential. Right now, it’s using a Hugo template that feels a bit outdated and has several issues where some features don’t work properly.

A fresh, modern, well-structured and visually appealing website makes onboarding easier for new developers by making documentation more accessible and understandable, also improves perception and trust in the framework by projecting a modern and reliable image, the site could make a difference in attracting new developers and improve overall developer experience. 🩵

My idea is to migrate the site and documentation to Astro, using the Starlight template, which is ideal for documentation. I’ve already started the migration and deployment, and it’s nearly complete. I just need to add a few small details and make a few minor corrections. Before submitting a pull request, I’d love to get feedback from the community and the project maintainers on this idea to ensure it aligns with everyone’s expectations. Plus,  I’d be happy to maintain the new site moving forward to keep it up to date! @appleboy 

**Fork**:
https://github.com/imlargo/gin-website

**Deploy** (Currently, my deployment is on Vercel, but it is compatible with GitHub Pages.)
https://gin-website.vercel.app/

![image](https://github.com/user-attachments/assets/30be29e6-3b14-4040-b216-36d25a927778)

![image](https://github.com/user-attachments/assets/c5a83e66-f711-4250-9126-602a438dc55a)

**Comment by austinheap** [comment]
This looks great, thank you for the contribution! One note: in dark mode the logo needs inverted or it looks wonky.

**Comment by imlargo** [comment]
> This looks great, thank you for the contribution! One note: in dark mode the logo needs inverted or it looks wonky.

@austinheap I added some white to the image to improve the appearance. However, I'm wondering if the outline might also need a color inversion. This is the current version—please let me know if you think any additional changes are needed.

![image](https://github.com/user-attachments/assets/c6011f35-0f54-4a9f-a711-fe75e89fbdc5)

**Comment by thinkerou** [comment]
nice!
thank you for the contribution!
if we change the new website, we should rename `gin-gonic.com` (only @javierprovecho ).
i also like the logo https://github.com/gin-gonic/logo/pull/6

**Comment by imlargo** [comment]
> i also like the logo [gin-gonic/logo#6](https://github.com/gin-gonic/logo/pull/6)

@thinkerou I've finished the remaining corrections, and I believe it's ready. I've just opened the [pull request](https://github.com/gin-gonic/website/pull/289). For now, I've kept the current logo as a placeholder until we have a final decision on the logo design. If needed, I can update it once we reach a consensus on which logo to use. Let me know if there are any further changes!

**Option 1:**
![image](https://github.com/user-attachments/assets/fb556f3d-576b-427a-8d77-b954a0133d7f)

**Option 2:**
![image](https://github.com/user-attachments/assets/79bd0998-f080-4b62-ac20-7fceaa229cba)

**Option 3:**
![image](https://github.com/user-attachments/assets/27efc123-5a81-40f4-ba96-4a4799578bb0)

**Comment by AdityaMayukhSom** [comment]
Any updates regarding merging this? A new polished look with better documentation will be really helpful as Gin seems to be one of the most widely used framework in Go ecosystem but lacks documentation and modern touch.

**Comment by Faizanahmedsy** [comment]
I'd be happy to assist with the UI design for this project. Let me know if there's any specific area you'd like me to focus on

**Comment by LuizWeitz** [comment]
Hi @imlargo, excellent work (amazing), we need to implement it quickly, because the current website today ends up "scaring" new developers when consulting it and especially the documentation! We need to attract more developers to continue strong with this Gin project!

**Comment by imlargo** [comment]
> Hi [@imlargo](https://github.com/imlargo), excellent work (amazing), we need to implement it quickly, because the current website today ends up "scaring" new developers when consulting it and especially the documentation! We need to attract more developers to continue strong with this Gin project!

That's right! Fortunately, the deployment is available for everyone, so everyone can read the docs in the new site if it's useful. However, I left a comment on the pull request since the documentation is probably a bit outdated. I'm still waiting for approval and a full review so I can move forward.

**Comment by appleboy** [comment]
Thank you very much @imlargo for your efforts, I will start reviewing the related PR.

**Comment by LuizWeitz** [comment]
Hi, @manucorporat, I think you would really enjoy looking at the work of the community, especially the @appleboy and @imlargo for new website of Gin.
